File: IdCaptureSurveyJava/app/src/main/java/com/scandit/datacapture/idcapturesurvey/home/HomeFragment.java
package com.scandit.datacapture.idcapturesurvey.home;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.scandit.datacapture.idcapturesurvey.navigation.NavigationUtils;
import com.scandit.datacapture.idcapturesurvey.scan.ScanFragment;
import com.scandit.datacapture.idcapturesurveyjava.databinding.FragmentHomeBinding;


public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentHomeBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.scanButton.setOnClickListener(v -> navigateToScanner());
    }

    private void navigateToScanner() {
        NavigationUtils.navigateToFragment(requireActivity(), ScanFragment.create(),
                ScanFragment.TAG);
    }
}

File: IdCaptureSurveyKotlin/app/src/main/java/com/scandit/datacapture/idcapturesurvey/home/HomeFragment.kt
package com.scandit.datacapture.idcapturesurvey.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.scandit.datacapture.idcapturesurvey.databinding.FragmentHomeBinding
import com.scandit.datacapture.idcapturesurvey.navigation.navigateToFragment
import com.scandit.datacapture.idcapturesurvey.scan.ScanFragment

class HomeFragment : Fragment() {

    private lateinit var binding: FragmentHomeBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
    ): View = FragmentHomeBinding.inflate(inflater, container, false)
        .also { binding = it }
        .root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.scanButton.setOnClickListener {
            navigateToScanner()
        }
    }

    private fun navigateToScanner() {
        requireActivity().navigateToFragment(ScanFragment.create(), ScanFragment.TAG)
    }
}